htlcswitch/link.go

```diff
@@ -797,12 +794,61 @@ func (l *channelLink) htlcManager() {
 	if l.cfg.SyncStates {
 		err := l.syncChanStates()
 		if err != nil {
-			l.errorf("unable to synchronize channel states: %v", err)
-			if err != ErrLinkShuttingDown {
-				// TODO(halseth): must be revisted when
-				// data-loss protection is in.
-				l.fail(LinkFailureError{code: ErrSyncError},
-					err.Error())
+			switch {
+			case err == ErrLinkShuttingDown:
+				log.Debugf("unable to sunc channel states, " +
+					"link is shutting down")
+				break
+
+			case err == lnwallet.ErrCommitSyncRemoteDataLoss:
+				// We failed syncing the commit chains,
+				// probably because the remote has lost
+				// state. We should force close the channel.
+				l.fail(
+					LinkFailureError{
+						code:       ErrSyncError,
+						ForceClose: true,
```

Comment:
Reminder that we should also record this instance such that we re-send the chan sync message each time we connect to the remote party until they sweep the output on chain. No reason this needs to be done in this current PR as it's already size-able, so we can just add a TODO here and an issue so we can proceed w/ review+testing of this PR.

Reply:
Added TODO, will create an issue to address this when merged.